netbox/dcim/forms/model_forms.py

```diff
@@ -1435,6 +1437,21 @@ class Meta:
             'mode': '802.1Q Mode',
         }
 
+    def clean_mac_address(self):
```

Comment:
Let's just ditch the `mac_address` field on both interface forms. There are numerous problems with trying to accommodate this workflow. It might be something to tackle in the future, but for now we should just consistently treat MAC addresses as individual from their assigned interfaces, the way we do with IP addresses.